File: Add-In/Scan/ReviveThis.Net.AddIn.Sample.CSharp/Classes/SampleResult.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Threading.Tasks;
using System.Windows.Forms;
using ReviveThis.Entities;
using ReviveThis.Enums;
using ReviveThis.Interfaces;
using ReviveThis.Structs;

namespace ReviveThis.Sample.CSharp.Classes
{
  public class SampleResult : IDetectionResultItem, IDetectionItemContextMenuCollection, ICustomAddInSection
  {
    private readonly ScanResultType _resultType;
    private readonly string _text;
    private const string SECTION_ID = "ZZ";

    public SampleResult(ScanResultType type, string text)
    {
      _resultType = type;
      _text = text;
    }

    public ScanResultType ResultType
    {
      get { return _resultType; }
    }

    public string LegacyString
    {
      get { return string.Format("{0} - {1}.", SECTION_ID, _text);  }
    }

    public ICollection<IDetectionItemContextMenu> MenuItems
    {
      get
      {
        var items = new Collection<IDetectionItemContextMenu>
        {
          new ScanItemContextMenu("Sample context menu", delegate
          {
            MessageBox.Show("Hello from your Add-On Sample context menu.", "Add-On Sample (C#)",
              MessageBoxButtons.OK, MessageBoxIcon.Information);
          })
        };

        return items;
      }
    }

    public CustomAddInSection CustomAddInSection
    {
      get { return new CustomAddInSection(SECTION_ID, "Wow I have a custom section..?"); }
    }

    public bool CanRepair
    {
      get { return true; }
    }

    public bool IsChecked
    {
      get { return true; }
    }

    public async Task<IDetectionRepairResult> Repair()
    {
      //await Task.FromResult(0);

      return new SampleRepairResult();

    }
  }
}

File: Add-In/Scan/ReviveThis.Net.AddIn.Sample.VisualBasic/Classes/SampleResult.vb
Imports System.Drawing
Imports System.Windows.Forms
Imports System.Collections.ObjectModel
Imports System.Runtime.Serialization
Imports ReviveThis.Entities
Imports ReviveThis.Enums
Imports ReviveThis.Interfaces
Imports ReviveThis.Structs
Imports System.Threading.Tasks

Namespace Classes

    Public Class SampleResult
        Implements IDetectionResultItem, IDetectionItemContextMenuCollection, ICustomAddInSection, IDetectionItemToolTip

        Private ReadOnly _resultType As ScanResultType
        Private ReadOnly _text As String
        Private Const SectionId As String = "ZZ"

        Public ReadOnly Property ResultType() As ScanResultType Implements IDetectionResultItem.ResultType
            Get
                Return _resultType
            End Get
        End Property

        Public ReadOnly Property LegacyString() As String Implements IDetectionResultItem.LegacyString
            Get
                Return String.Format("{0} - {1}.", SectionId, _text)
            End Get
        End Property

        Public ReadOnly Property ToolTip() As String Implements IDetectionItemToolTip.ToolTip
            Get
                Return "Add-On ToolTip Sample (Visual Basic)"
            End Get
        End Property

        Public Sub New(type As ScanResultType, text As String)
            _resultType = type
            _text = text
        End Sub

        Public ReadOnly Property MenuItems() As ICollection(Of IDetectionItemContextMenu) Implements IDetectionItemContextMenuCollection.MenuItems
            Get
                Dim items As New Collection(Of IDetectionItemContextMenu)
                items.Add(New ScanItemContextMenu("Sample context menu", AddressOf Test))
                Return items
            End Get
        End Property

        Private Shared Sub Test(sender As Object, e As EventArgs)
            MessageBox.Show("Hello from your Add-On Sample context menu.", "Add-On Sample (Visual Basic)", MessageBoxButtons.OK, MessageBoxIcon.Information)
        End Sub

        Public ReadOnly Property CustomAddInSection() As CustomAddInSection Implements ICustomAddInSection.CustomAddInSection
            Get
                Return New CustomAddInSection(SectionId, "Wow I have a custom section..?")
            End Get
        End Property

        Public ReadOnly Property CanRepair() As Boolean Implements IDetectionRepair.CanRepair
            Get
                Return True
            End Get
        End Property

        Public ReadOnly Property IsChecked() As Boolean Implements IDetectionRepair.IsChecked
            Get
                Return False
            End Get
        End Property

        Public Function Repair() As Task(Of IDetectionRepairResult) Implements IDetectionRepair.Repair
            Return Nothing
        End Function
    End Class
End Namespace